### PR TITLE
NftDetectionController: providerConfig -> selectedNetworkClientId

### PR DESCRIPTION
**Pull request by mcmire**
## Explanation

The `providerConfig` state property is being removed from NetworkController. Currently this property is used in NftDetectionController to get the currently selected chain, but `selectedNetworkClientId` can be used instead. This commit makes that transition so that we can fully drop `providerConfig`.

## References

Progresses #4185.

## Changelog

(Updates to changelog in PR.)

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/assets-controllers/src/NftDetectionController.test.ts [hunk 1]
```diff
@@ -378,13 +480,107 @@ describe('NftDetectionController', () => {
     });
   });
 
+  it('should respond to chain ID changing when using legacy polling', async () => {
```
Comment (mcmire) [inline packages/assets-controllers/src/NftDetectionController.test.ts:483]: This was a missing test.

File: packages/assets-controllers/src/NftDetectionController.test.ts [hunk 2]
```diff
@@ -378,13 +480,107 @@ describe('NftDetectionController', () => {
...
+
   it('should detect and add NFTs correctly when blockaid result is not included in response', async () => {
     const mockAddNft = jest.fn();
     await withController(
       { options: { addNft: mockAddNft } },
-      async ({ controller, triggerPreferencesStateChange }) => {
+      async ({ controller, controllerEvents }) => {
         const selectedAddress = '0x1';
-        triggerPreferencesStateChange({
+        controllerEvents.preferencesStateChange({
```
Comment (mcmire) [inline packages/assets-controllers/src/NftDetectionController.test.ts:583]: Note that there are a lot of changes in this PR that are unrelated to the change from `providerConfig` to `selectedNetworkClientId`. I took this opportunity to align the tests in this file to use our latest convention for mocking events couched as callbacks.

File: packages/assets-controllers/src/NftDetectionController.test.ts [hunk 3]
```diff
@@ -871,33 +1074,35 @@ type WithControllerArgs<ReturnValue> =
...
       chainId: ChainId.mainnet,
       onNftsStateChange: (listener) => {
-        nftStateChangeListeners.push(listener);
+        controllerEvents.nftsStateChange = listener;
       },
       onPreferencesStateChange: (listener) => {
-        preferencesStateChangeListeners.push(listener);
+        controllerEvents.preferencesStateChange = listener;
+      },
+      onNetworkStateChange: (listener) => {
```
Comment (mcmire) [inline packages/assets-controllers/src/NftDetectionController.test.ts:1103]: Previously we were just passing `jest.fn()` as the value for this callback, but now we are capturing the listener so that we can call it in tests, effectively firing the event.

File: packages/assets-controllers/src/NftDetectionController.test.ts [hunk 4]
```diff
@@ -264,7 +276,6 @@ describe('NftDetectionController', () => {
           },
         ],
       });
-    console.log(nock.activeMocks());
```
Comment (mcmire) [inline packages/assets-controllers/src/NftDetectionController.test.ts:267]: I am not sure why we were doing this; this was just adding to the noise when running these tests.

Comment (cryptodev-2s) [reply packages/assets-controllers/src/NftDetectionController.test.ts:267]: I guess someone forgot to remove it while debugging

**Review by cryptodev-2s — APPROVED** [review]
LGTM!
